src/ddscxx/include/dds/core/detail/inttypes.hpp

```diff
@@ -23,7 +23,12 @@
  * simply include that one. Under toolchains that do not, this header must
  * provide equivalent definitions.
  */
-#include <stdint.h>
+#if defined(__GNUC__) && __GNUC__ < 5
+#define __STDC_FORMAT_MACROS
+#endif
+
+//#include <stdint.h>
```

Comment:
I am not sure whether `inttypes.h` actually implies `stdint.h`. It seems no more than reasonable, but including both seems a practical of avoiding any problems.
```suggestion
#include <stdint.h>
```